src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala

```diff
@@ -476,9 +476,20 @@ trait NamesDefaults { self: Analyzer =>
   def defaultGetter(param: Symbol, context: Context): Symbol = {
     val i = param.owner.paramss.flatten.indexWhere(p => p.name == param.name) + 1
     if (i > 0) {
-      val defGetterName = nme.defaultGetterName(param.owner.name, i)
-      if (param.owner.isConstructor) {
-        val mod = companionSymbolOf(param.owner.owner, context)
+      def isScala3SyntheticApply(meth: Symbol): Boolean = {
+        // According to rules in Scala 3, a synthetic method named `apply`
+        // should use `<init>` as the prefix of its default getters,
+        // i.e. reuse the constructor's default getters.
+        meth.isScala3Defined && meth.isSynthetic && meth.name == nme.apply
```

Comment:
In addition to `meth.isSynthetic` you add something like `meth.owner.isModule && companionSymbolOf(meth.owner).isCase`. But it's probably unlikely that there are other synthetic `apply` methods, so as you prefer.

Reply:
I considered adding that, however the rule in Scala 3 is only a synthetic apply method - so perhaps a synthetic apply method forever more implies being in the companion object

Reply:
in Scala 3 this synthetic apply rule also works for the `universal apply methods` (which is backed up by a real `apply` method symbol) to it does not need to check for `case`